app/scripts/lib/extension-update-manager.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Manages extension update detection and application.
+ * Provides a clean interface for detecting and applying extension updates
+ * during idle periods to minimize user disruption.
+ */
+
+import browser from 'webextension-polyfill';
+
+export class ExtensionUpdateManager {
+  private updatePending = false;
+
+  private isIdle = false;
+
+  /**
+   * Initializes the update manager by setting up event listeners
+   * for update detection.
+   */
+  public initialize(): void {
+    // Ensure we start in a non-idle state during initialization
+    this.isIdle = false;
+
+    browser.runtime.onUpdateAvailable.addListener(
+      this.handleUpdateAvailable.bind(this),
+    );
+  }
+
+  /**
+   * Sets the idle state of the extension.
+   *
+   * @param idle - Whether the extension is currently idle
+   */
+  public setIdleState(idle: boolean): void {
+    this.isIdle = idle;
+
+    // If we're now idle and there's a pending update, apply it immediately
+    if (idle && this.updatePending) {
```

Comment:
Nit: We're checking these two flags redundantly here, both inside and outside `applyPendingUpdateIfNeeded`.

Maybe we could remove the checks here, and leave it to `applyPendingUpdateIfNeeded` to check? Or we could keep them here, and rename `applyPendingUpdateIfNeeded` to `applyPendingUpdate`, removing the internal checks.

Reply:
Likewise for the `isIdle` check in `handleUpdateAvailable`

Reply:
🕵🏻‍♂️ good catch. renamed and removed redundant logic 79bb9e7f9ff95381a32da33de39cc56c6869a534